TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeychainQuery.swift

```diff
@@ -47,68 +47,111 @@ protocol KeychainQueryProtocol {
+    properties([
+      kSecClass: kSecClassKey,
+      kSecAttrLabel: alias,
+      kSecValueRef: key,
+      kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    ], saving: true)
   }
   
   func deleteKey(withAlias alias: String) -> Query {
-    [kSecClass: kSecClassKey,
+    properties([
+     kSecClass: kSecClassKey,
      kSecAttrLabel: alias,
-     kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly] as Query
+     kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    ])
   }
-  
+
   func save(data: Data, withKey key: String, withServiceName service: String?) -> Query {
     var query = [kSecClass: kSecClassGenericPassword,
      kSecAttrAccount: key,
      kSecValueData: data,
      kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly] as Query
     if let service = service {
-      query[kSecAttrService as String] = service
+      query[kSecAttrService] = service
     }
-    return query
+    return properties(query, saving: true)
   }
-  
+
   func getData(withKey key: String) -> Query {
-    [kSecClass: kSecClassGenericPassword,
-     kSecAttrAccount: key,
-     kSecReturnData: true,
-     kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly] as Query
+    properties([
+      kSecClass: kSecClassGenericPassword,
+      kSecAttrAccount: key,
+      kSecReturnData: true,
+      kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    ])
   }
-  
+
   func getAll(withServiceName service: String?) -> Query {
     var query = [kSecClass: kSecClassGenericPassword,
      kSecReturnAttributes: true,
      kSecReturnData: true,
      kSecMatchLimit: kSecMatchLimitAll,
      kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly] as Query
     if let service = service {
-      query[kSecAttrService as String] = service
+      query[kSecAttrService] = service
     }
-    return query
+    return properties(query)
   }
-  
+
   func delete(withKey key: String) -> Query {
-    [kSecClass: kSecClassGenericPassword,
-     kSecAttrAccount: key] as Query
+    properties([
+      kSecClass: kSecClassGenericPassword,
+      kSecAttrAccount: key
+    ])
   }
-  
+
   func deleteItems(withServiceName service: String?) -> Query {
     var query = [kSecClass: kSecClassGenericPassword] as Query
     if let service = service {
-      query[kSecAttrService as String] = service
+      query[kSecAttrService] = service
     }
-    return query
+    return properties(query)
+  }
+
+  // MARK: - Private Methods
+
+  func properties(_ properties: Query, saving: Bool = false) -> Query {
+    var properties = properties
+
+    if let accessGroup = accessGroup {
+      properties[kSecAttrAccessGroup] = accessGroup
+    }
+
+    if let synchronizable = synchronizable, synchronizable {
+      if saving {
+        properties[kSecAttrSynchronizable] = kCFBooleanTrue
+      } else {
+        properties[kSecAttrSynchronizable] = kSecAttrSynchronizableAny
```

Comment:
Do we need this for other queries like get and delete? Will it be enough only for saving?

Reply:
Yes, is needed, if for any reason we are using synchronizable on saving, we must use `kSecAttrSynchronizableAny` in get & delete queries, otherwise it won't perform any query in data with `kSecAttrSynchronizable` in `true`. That behaviour is also described in the documentation attached to the PR.